log4j-1.2-api/src/main/java/org/apache/log4j/Category.java

```diff
@@ -376,7 +376,7 @@ public Appender getAppender(final String name) {
         if (appender == null && LogManager.isLog4jCorePresent()) {
             final org.apache.logging.log4j.core.Appender coreAppender = CategoryUtil.getAppenders(logger).get(name);
             if (coreAppender != null) {
-                addAppender(appender = new AppenderWrapper(coreAppender));
+                addAppender(appender = AppenderWrapper.wrap(coreAppender));
```

Comment:
This is the main change: if `coreAppender` is an `AppenderAdapter` and **no** filters are configured, then the wrapped object is returned.

@garydgregory: can you check if this is Ok?